lib/matplotlib/_mathtext.py

```diff
@@ -1817,8 +1817,11 @@ def __init__(self):
         def set_names_and_parse_actions():
             for key, val in vars(p).items():
                 if not key.startswith('_'):
-                    # Set names on everything -- very useful for debugging
-                    val.setName(key)
+                    # Set names on (almost) everything -- very useful for debugging
+                    # token, placeable, and auto_delim are forward references which
+                    # are left without names to ensure useful error messages
+                    if key not in ("token", "placeable", "auto_delim"):
+                        val.setName(key)
```

Comment:
This change I didn't fully intend on keeping in this PR... this was part of the discussion from #26152 that is likely to be part of the fix for making the error messages work for future versions of pyparsing...

I'm not totally against leaving this in as I think it is likely to be wanted soon, but will allow reviewers to revert it if it is too soon.

```suggestion
                    # Set names on everything -- very useful for debugging
                    val.setName(key)
```